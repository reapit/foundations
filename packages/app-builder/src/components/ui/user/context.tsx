import { usePageId } from '@/components/hooks/use-page-id'
import React from 'react'

const Context = () => {
  const { context } = usePageId()

  return <div>{JSON.stringify(context, null, '\t')}</div>
}

export default Context
