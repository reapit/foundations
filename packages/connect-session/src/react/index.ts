import React from 'react'
import { ReapitConnectHook, ReapitConnectSession } from '../types'
import { ReapitConnectBrowserSession } from '../browser'

export const useReapitConnect = (reapitConnectBrowserSession: ReapitConnectBrowserSession): ReapitConnectHook => {
  const [connectSession, setConnectSession] = React.useState<ReapitConnectSession | null>(null)

  React.useEffect(() => {
    const connectGetSession = async () => {
      const session = await reapitConnectBrowserSession.connectSession()
      if (session) {
        setConnectSession(session)
      }
    }

    connectGetSession()
  }, [])

  const connectAuthorizeRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectAuthorizeRedirect()
  }, [])

  const connectLoginRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLoginRedirect()
  }, [])

  const connectLogoutRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLogoutRedirect()
  }, [])

  const connectIsDesktop = reapitConnectBrowserSession.connectIsDesktop

  const connectHasSession = reapitConnectBrowserSession.connectHasSession

  return {
    connectSession,
    connectAuthorizeRedirect,
    connectLoginRedirect,
    connectLogoutRedirect,
    connectIsDesktop,
    connectHasSession,
  }
}
