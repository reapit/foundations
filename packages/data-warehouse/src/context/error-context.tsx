import React, { createContext, Dispatch, SetStateAction, useState } from 'react'
import { ErrorData } from '@reapit/elements'

export interface ErrorContextProps {
  componentErrorState: ErrorData | null
  serverErrorState: ErrorData | null
  setComponentErrorState: Dispatch<SetStateAction<ErrorData | null>>
  setServerErrorState: Dispatch<SetStateAction<ErrorData | null>>
}

export const ErrorContext = createContext<ErrorContextProps>({} as ErrorContextProps)

const { Consumer, Provider } = ErrorContext

export const ErrorProvider: React.FC = ({ children }) => {
  const [componentErrorState, setComponentErrorState] = useState<ErrorData | null>(null)
  const [serverErrorState, setServerErrorState] = useState<ErrorData | null>(null)

  return (
    <Provider value={{ componentErrorState, setComponentErrorState, serverErrorState, setServerErrorState }}>
      {children}
    </Provider>
  )
}

export const ErrorConsumer = Consumer
