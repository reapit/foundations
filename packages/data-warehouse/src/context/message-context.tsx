import React, { createContext, Dispatch, SetStateAction, useState } from 'react'
import { ToastVariant } from '@reapit/elements'

export interface MessageState {
  visible: boolean
  message: string
  variant: ToastVariant
}

export interface MessageContextProps {
  messageState: MessageState
  setMessageState: Dispatch<SetStateAction<MessageState>>
}

export const MessageContext = createContext<MessageContextProps>({} as MessageContextProps)

const { Consumer, Provider } = MessageContext

export const MessageProvider: React.FC = ({ children }) => {
  const [messageState, setMessageState] = useState<MessageState>({
    variant: 'primary',
    message: '',
    visible: false,
  })

  return <Provider value={{ messageState, setMessageState }}>{children}</Provider>
}

export const MessageConsumer = Consumer
