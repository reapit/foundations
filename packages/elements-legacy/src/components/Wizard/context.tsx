import React, { useContext, createContext } from 'react'
import { FormikProps } from 'formik'
import { WizardStepProps } from '.'

export type WizardContextValues = {
  current: string
  goNext: () => void
  goPrev: () => void
  goTo: (stepId: string) => void
  close: () => void
  currentIndex: number
  isFirst: boolean
  isLast: boolean
  steps: WizardStepProps<any>[]
  isLoading: boolean
}

// @ts-ignore: ignore fulfill default values
const WizardContext = createContext<WizardContextValues>({})

export const WizardContextProvider: React.FC<{ value: WizardContextValues }> = ({ children, value }) => (
  <WizardContext.Provider value={value}>{children}</WizardContext.Provider>
)

export const useWizardContext = (): WizardContextValues => {
  return useContext(WizardContext)
}

export type WizardRenderCallbackParams = {
  context: WizardContextValues
  form: FormikProps<any>
}

export type WizardRenderContextValues = {
  leftFooterRender: ((params: WizardRenderCallbackParams) => React.ReactNode) | React.ReactNode
  rightFooterRender: ((params: WizardRenderCallbackParams) => React.ReactNode) | React.ReactNode
}

// @ts-ignore: ignore fulfill default values
const RenderWizardContext = createContext<WizardRenderContextValues>({})

export const WizardRenderContextProvider = ({
  children,
  value,
}: {
  value: WizardRenderContextValues
  children: React.ReactNode
}) => <RenderWizardContext.Provider value={value}>{children}</RenderWizardContext.Provider>

export const useRenderWizardContext = (): WizardRenderContextValues => {
  return useContext(RenderWizardContext)
}
