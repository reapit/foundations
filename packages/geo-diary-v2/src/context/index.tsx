export { default as AuthContext } from './auth-context'
