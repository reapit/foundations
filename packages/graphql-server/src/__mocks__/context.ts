import { ServerContext } from '../index'

export const mockContext = {
  traceId: 'mockTraceId',
  authorization: 'mockAuthorization',
} as ServerContext
