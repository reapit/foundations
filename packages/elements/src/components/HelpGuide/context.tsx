import * as React from 'react'
import { HelpGuideStepProps } from '.'

function createContext<A>() {
  const ctx = React.createContext<A | undefined>(undefined)
  function useCtx() {
    const c = React.useContext(ctx)
    if (!c) throw new Error('useContext must be inside a Provider with a value')
    return c
  }
  return [useCtx, ctx.Provider] as const
}

export type HelpGuideContextValues = {
  current: string
  goNext: () => void
  goPrev: () => void
  goTo: (stepIndex: number) => void
  currentIndex: number
  isFirst: boolean
  isLast: boolean
  steps: HelpGuideStepProps[]
  isLoading: boolean
}

export const [useHelpGuideContext, HelpGuideContextProvider] = createContext<HelpGuideContextValues>()
