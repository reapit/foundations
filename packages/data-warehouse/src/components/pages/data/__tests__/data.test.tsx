import * as React from 'react'
import { mount } from 'enzyme'
import Data from '../data'
import { MessageProvider } from '../../../../context/message-context'

// jest.mock('../../../../services/shares')
// jest.mock('../../../../services/data-sets')
// jest.mock('../../../../services/subscriptions')

jest.mock('@reapit/connect-session', () => ({
  ReapitConnectBrowserSession: jest.fn(),
  useReapitConnect: () => ({
    connectSession: {
      loginIdentity: {
        developerId: 'SOME_ID',
      },
    },
  }),
}))

describe('Data', () => {
  it('should match a snapshot', () => {
    expect(
      mount(
        <MessageProvider>
          <Data />
        </MessageProvider>,
      ),
    ).toMatchSnapshot()
  })
})
