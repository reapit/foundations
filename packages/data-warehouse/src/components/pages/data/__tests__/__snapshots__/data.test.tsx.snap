// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data should match a snapshot 1`] = `
<Component>
  <Component>
    <Component>
      <div
        className=""
        data-test=""
      >
        <Component
          isHeadingSection={true}
        >
          <h3
            className=""
            id=""
          >
            Data
          </h3>
        </Component>
        <Component>
          <section
            className=""
            data-test=""
          >
            <Component>
              <h5
                className=""
                id=""
              >
                Available Data
              </h5>
            </Component>
            <Component>
              <div
                className="loader-spinner body"
                data-test=""
              >
                <div />
                <div />
                <div />
                <div />
              </div>
            </Component>
          </section>
        </Component>
        <Component>
          <section
            className=""
            data-test=""
          >
            <Component>
              <h5
                className=""
                id=""
              >
                Data Shares
              </h5>
            </Component>
            <Component>
              <div
                className="loader-spinner body"
                data-test=""
              >
                <div />
                <div />
                <div />
                <div />
              </div>
            </Component>
          </section>
        </Component>
      </div>
    </Component>
  </Component>
</Component>
`;
